File: checkers/190/main.go
package main

import (
	"fmt"
	"testlib"
	"testlib/core"
)

func createMatrix0(n int, m int) [][]int {
	matrix := make([][]int, n)
	for i := 0; i < n; i++ {
		matrix[i] = make([]int, m)
		for j := 0; j < m; j++ {
			matrix[i][j] = 0
		}
	}
	return matrix
}

func checkValueOutOfSegmen(val int, a int, b int) bool {
	if val < a || val > b {
		return true
	} else {
		return false
	}
}

func main() {
	defer func() {
		_ = testlib.GetStream(testlib.TaskOutput).Close()
		_ = testlib.GetStream(testlib.InputFromUser).Close()
	}()
	inputStream := testlib.GetReader(testlib.TaskInput)
	outputStream := testlib.GetReader(testlib.TaskOutput)
	inputFromUserStream := testlib.GetReader(testlib.InputFromUser)

	n := core.ReadInt(inputStream)
	g := createMatrix0(n, n)
	for i := 0; i < n; i++ {
		for j := 0; j < n; j++ {
			g[i][j] = core.ReadInt(inputStream)
		}
	}
	res := core.ReadWord(inputFromUserStream)
	ans := core.ReadWord(outputStream)
	if res != ans {
		testlib.WriteAnswer(fmt.Sprintf("NOPE"), testlib.WA)
		return
	}
	if res == "NO" {
		testlib.WriteAnswer("OK", testlib.OK)
		return
	}
	cnt1 := core.ReadInt(inputFromUserStream)
	if checkValueOutOfSegmen(cnt1, 2, 2*n) {
		testlib.WriteAnswer(fmt.Sprintf("Value out of segment %d and %d", 2, 2*n), testlib.WA)
		return
	}
	x := core.ReadInt(inputFromUserStream)
	if checkValueOutOfSegmen(x, 1, n) {
		testlib.WriteAnswer(fmt.Sprintf("Value out of segment %d and %d", 1, n), testlib.WA)
		return
	}
	cnt := 0
	y := 0
	for i := 0; i < cnt1-1; i++ {
		y = core.ReadInt(inputFromUserStream)
		if checkValueOutOfSegmen(y, 1, n) {
			testlib.WriteAnswer(fmt.Sprintf("Value out of segment %d and %d", 1, n), testlib.WA)
			return
		}
		if g[x-1][y-1] == 100000 {
			testlib.WriteAnswer(fmt.Sprintf("WA two unconnected vertices  %d %d", x, y), testlib.WA)
			return
		}
		cnt += g[x-1][y-1]
		x = y
	}
	if cnt >= 0 {
		testlib.WriteAnswer(fmt.Sprintf("ABOVE ZERO"), testlib.WA)
		return
	}
	testlib.WriteAnswer("OK", testlib.OK)
}

File: checkers/190/main.cpp
#include <checkers/testlib.h>
#include <iostream>
#include <vector>
#include <algorithm>
#include <string>

using namespace NTestlib;

int main(int argc, char *argv[])
{
    InitChecker(argc, argv);
    int n = File.ReadInt();
    //
    std::vector<std::vector<int>> 
    g(n, std::vector<int>(n, 0)); // одна строка
    for (int i = 0; i < n; i++)
        for (int j = 0; j < n; j++)
            g[i][j] = File.ReadInt();
    std::string res = Ans.ReadWord();
    std::string ans = Out.ReadWord();
    if (res != ans)
        QuitWith(WA, "NOPE");
    if (res == "NO")
        QuitWith(AC, "OK");
    int cnt1 = Out.ReadInt(2, 2 * n);
    int x = Out.ReadInt(1, n);
    int cnt = 0;
    for (int i = 0; i < cnt1 - 1; i++)
    {
        int y = Out.ReadInt(1, n);
        if (g[x - 1][y - 1] == 100000)
        {
            std::stringstream msg;
            //
            msg << 
            "WA two unconnected vertices " << 
            x << " " << y; // одна строка
            QuitWith(WA, msg.str());
        }
        cnt += g[x - 1][y - 1];
        x = y;
    }
    if (cnt >= 0)
        QuitWith(WA, "ABOVE ZERO");
    QuitWith(AC, "OK");
}